ext/mysql2/result.c

```diff
@@ -169,6 +169,160 @@ static VALUE rb_mysql_result_fetch_field(VALUE self, unsigned int idx, int symbo
   return rb_field;
 }
 
+static VALUE rb_mysql_result_fetch_field_type(VALUE self, unsigned int idx) {
+  VALUE rb_field_type;
+  GET_RESULT(self);
+
+  if (wrapper->fieldTypes == Qnil) {
+    wrapper->numberOfFields = mysql_num_fields(wrapper->result);
+    wrapper->fieldTypes = rb_ary_new2(wrapper->numberOfFields);
+  }
+
+  unsigned long mb_per_char = 3;
+
+  rb_field_type = rb_ary_entry(wrapper->fieldTypes, idx);
+  if (rb_field_type == Qnil) {
+    MYSQL_FIELD *field = NULL;
+    rb_encoding *default_internal_enc = rb_default_internal_encoding();
+    rb_encoding *conn_enc = rb_to_encoding(wrapper->encoding);
+
+    field = mysql_fetch_field_direct(wrapper->result, idx);
+
+    switch(field->type) {
+      case MYSQL_TYPE_NULL:         // NULL
+        rb_field_type = rb_str_new_cstr("null");
+        break;
+      case MYSQL_TYPE_TINY:         // signed char
+        rb_field_type = rb_sprintf("tinyint(%ld)", field->length);
+        break;
+      case MYSQL_TYPE_SHORT:        // short int
+        rb_field_type = rb_sprintf("smallint(%ld)", field->length);
+        break;
+      case MYSQL_TYPE_YEAR:         // short int
+        rb_field_type = rb_sprintf("year(%ld)", field->length);
+        break;
+      case MYSQL_TYPE_INT24:        // int
+        rb_field_type = rb_sprintf("mediumint(%ld)", field->length);
+        break;
+      case MYSQL_TYPE_LONG:         // int
+        rb_field_type = rb_sprintf("int(%ld)", field->length);
+        break;
+      case MYSQL_TYPE_LONGLONG:     // long long int
+        rb_field_type = rb_sprintf("bigint(%ld)", field->length);
+        break;
+      case MYSQL_TYPE_FLOAT:        // float
+        rb_field_type = rb_sprintf("float(%ld,%d)", field->length, field->decimals);
+        break;
+      case MYSQL_TYPE_DOUBLE:       // double
+        rb_field_type = rb_sprintf("double(%ld,%d)", field->length, field->decimals);
+        break;
+      case MYSQL_TYPE_TIME:         // MYSQL_TIME
+        rb_field_type = rb_str_new_cstr("time");
+        break;
+      case MYSQL_TYPE_DATE:         // MYSQL_TIME
+      case MYSQL_TYPE_NEWDATE:      // MYSQL_TIME
+        rb_field_type = rb_str_new_cstr("date");
+        break;
+      case MYSQL_TYPE_DATETIME:     // MYSQL_TIME
+        rb_field_type = rb_str_new_cstr("datetime");
+        break;
+      case MYSQL_TYPE_TIMESTAMP:    // MYSQL_TIME
+        rb_field_type = rb_str_new_cstr("timestamp");
+        break;
+      case MYSQL_TYPE_DECIMAL:      // char[]
+      case MYSQL_TYPE_NEWDECIMAL:   // char[]
+        rb_field_type = rb_sprintf("decimal(%ld,%d)", field->length - 2, field->decimals);
+        break;
+      case MYSQL_TYPE_STRING:       // char[]
+        if (field->flags & ENUM_FLAG) {
+          rb_field_type = rb_str_new_cstr("enum");
+        } else if (field->flags & SET_FLAG) {
+          rb_field_type = rb_str_new_cstr("set");
+        } else {
+          if (field->charsetnr == 63) {
```

Comment:
From the mysql documentation. Will change this to a constant later.
> To distinguish between binary and nonbinary data for string data types, check whether the charsetnr value is 63. If so, the character set is binary, which indicates binary rather than nonbinary data. This enables you to distinguish BINARY from CHAR, VARBINARY from VARCHAR, and the BLOB types from the TEXT types.

Reply:
Please either add this comment into the code here, or break out `63` as a constant and put the comment next to it.